include/osquery/http_client.h

```diff
@@ -0,0 +1,438 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#pragma once
+
+#include <boost/asio.hpp>
+#include <boost/asio/deadline_timer.hpp>
+#include <boost/asio/ssl.hpp>
+#include <boost/beast/core.hpp>
+#include <boost/beast/http.hpp>
+#include <boost/optional/optional.hpp>
+
+// clang-format off
+#ifdef WIN32
+#pragma warning(push, 3)
+
+/*
+ * Suppressing warning C4005:
+ * 'ASIO_ERROR_CATEGORY_NOEXCEPT': macro redefinition
+ */
+#pragma warning(disable: 4005)
+
+/*
+ * Suppressing warning C4244:
+ * 'argument': conversion from '__int64' to 'long', possible loss of data
+ */
+#pragma warning(disable: 4244)
+#endif
+
+#include <boost/network/uri.hpp>
+#include <boost/network/uri/uri_io.hpp>
+
+#ifdef WIN32
+#pragma warning(pop)
+
+/// We need to reinclude this to re-enable boost's warning suppression
+#include <boost/config/compiler/visualc.hpp>
+#endif
+// clang-format on
+
+#include <openssl/crypto.h>
+#include <openssl/ssl.h>
+
+#ifndef OPENSSL_NO_SSL2
+#define OPENSSL_NO_SSL2 1
+#endif
+
+#ifndef OPENSSL_NO_SSL3
+#define OPENSSL_NO_SSL3 1
+#endif
+
+#define OPENSSL_NO_MD5 1
+#define OPENSSL_NO_DEPRECATED 1
+
+/// Newer versions of LibreSSL will lack SSL methods.
+extern "C" {
+#if defined(NO_SSL_TXT_SSLV3)
+SSL_METHOD* SSLv3_server_method(void);
+SSL_METHOD* SSLv3_client_method(void);
+SSL_METHOD* SSLv3_method(void);
+#endif
+void ERR_remove_state(unsigned long);
+}
+
+namespace boost_system = boost::system;
+namespace boost_asio = boost::asio;
+namespace beast_http = boost::beast::http;
+
+namespace osquery {
+namespace http {
+
+typedef boost_asio::ssl::stream<boost_asio::ip::tcp::socket&> ssl_stream;
+typedef beast_http::request<beast_http::string_body> beast_http_request;
```

Comment:
Consider `empty_body` instead of `string_body`, since the requests never have a payload (or do they?)